src/util/expr.cpp

```diff
@@ -287,6 +287,65 @@ void exprt::visit_post(std::function<void(const exprt &)> visitor) const
   visit_post_template(visitor, this);
 }
 
+optionalt<exprt>
+exprt::transform_pre(std::function<optionalt<exprt>(exprt)> visitor) const
+{
+  auto visitor_result = visitor(*this);
+
+  exprt tmp;
+
+  if(visitor_result.has_value())
+    tmp = visitor_result.value();
+  else
+    tmp = *this;
```

Comment:
How about `exprt tmp = visitor_result.value_or(*this);`

Reply:
sure